kafka/mockcluster.go

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka
+
+import "C"
+import "unsafe"
+
+/*
+#include <stdlib.h>
+#include "select_rdkafka.h"
+#include "glue_rdkafka.h"
+*/
+import "C"
+
+type MockCluster struct {
+	rk       *C.rd_kafka_t
+	mcluster *C.struct_rd_kafka_mock_cluster_s
+	cConf    *C.rd_kafka_conf_t
+}
+
+/*	NewMockCluster provides a mock Kafka cluster with a configurable
```

Comment:
nitty, but it is important for doc-generation: there should be a single whitespace between `*` and the first character in each line, like so:
```go
/* NewMockCluster provides a ..
 * number of brokers ..
 *  operations, error injec...
```